python/test/backend/test_stages_override.py

```diff
@@ -0,0 +1,98 @@
+import os
```

Comment:
please make the test more unitary, meaning test what you want to enforce rather than some specific use case. It looks like you are implementing your whole use case in the test. 
Ideally we want to decide on an interface and leave the implementation up to downstream users.

Reply:
I’m ok with downstream having to fix but it might be helpful if changes to the test could show more obviously how to do said fix?

Reply:
Hello @ThomasRaoux, I have some the cleanup of the tests to drop all the fundamentally unnecessary implementation bits. Let me know if more is needed to be cleaned up.